### ISSUE TITLE
setStatusBarMessage completion handler

### ISSUE THREAD

**Issue by DifficultNick** [Microsoft/vscode#56369]
It would be nice to have a handler to subscribe the status bar redrawn event. Because now I need to use `setTimeout` to be sure that message was shown before my extension does next steps.

**Comment by jrieken** [comment]
> Because now I need to use setTimeout to be sure that message was shown before my extension does next steps.

Why that? There is no delay, just some debouncing for rapidly repeated calls. When do you not see a status bar message?

**Comment by DifficultNick** [comment]
Here is an example:
```typescript
async function compute(): Promise<string>
{
    let res = "";
    for (let i = 1; i < 999999; i++)
        for (let j = 1; j < 9999999; i++)
            res += (i++) * (j++);
    return res;
}

export function test()
{
    console.log('started');
    vscode.window.setStatusBarMessage("computing...");
    compute().then(x => console.log(x.length));
}
```

I see `computing...` message only after `compute()` completed

**Comment by jrieken** [comment]
Yeah, but only because `compute` overloads the event loop. Internally the `setStatusBarMessage`-function is debounces with `setTimeout(fn, 0)`. That means the `compute` function steels all resources so that we cannot can the message to the renderer. Also all other computation is block and I hope your extension doesn't actually do something like that.
